File: backend-python/app/schemas/weekly_report.py

import json
from pydantic import BaseModel, field_validator


def _parse_images(v):
    if isinstance(v, str):
        try:
            parsed = json.loads(v)
            if isinstance(parsed, list):
                return parsed
        except (json.JSONDecodeError, TypeError):
            pass
        return []
    return v


class WeeklyReportCreate(BaseModel):
    report_id: str | None = None
    project_id: str | None = None
    project_name: str | None = None
    week_start_date: str | None = None
    week_end_date: str | None = None
    report_date: str
    work_summary: str | None = None
    work_content: str | None = None
    next_week_plan: str | None = None
    issues: str | None = None
    suggestions: str | None = None
    images: list[str] | None = None
    manager_signature: str | None = None

    @field_validator('images', mode='before')
    @classmethod
    def validate_images(cls, v):
        return _parse_images(v)


class WeeklyReportUpdate(BaseModel):
    project_id: str | None = None
    project_name: str | None = None
    week_start_date: str | None = None
    week_end_date: str | None = None
    report_date: str | None = None
    work_summary: str | None = None
    work_content: str | None = None
    next_week_plan: str | None = None
    issues: str | None = None
    suggestions: str | None = None
    images: list[str] | None = None
    manager_signature: str | None = None
    status: str | None = None

    @field_validator('images', mode='before')
    @classmethod
    def validate_images(cls, v):
        return _parse_images(v)


class WeeklyReportApprove(BaseModel):
    approved: bool
    reject_reason: str | None = None


class WeeklyReportSign(BaseModel):
    manager_signature: str

File: H5/src/types/models/weeklyReport.ts
/**
 * 周报数据模型
 */

/**
 * 周报查询参数
 */
export interface WeeklyReportQueryParams {
  page?: number
  size?: number
  project_name?: string
  report_date?: string
  status?: string
  created_by?: string
}

/**
 * 创建周报请求
 */
export interface WeeklyReportCreate {
  project_id?: string
  project_name?: string
  report_date?: string
  report_id?: string
  work_summary?: string
  next_week_plan?: string
  issues?: string
  suggestions?: string
  report_content?: string
}

/**
 * 更新周报请求
 */
export interface WeeklyReportUpdate {
  report_content?: string
  status?: string
}
